### PR TITLE
tiltfile: store all manifest names on tiltfile [ch776]

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @nicks, @jazzdan,

Please review the following commits I made in branch maiamcc/tiltfile-knows-all-manifests:

168497bcd16635bc97b9cc7a752f267b56f8a76f (2018-11-12 18:02:13 -0500)
tiltfile: store all manifest names on tiltfile

480f7e3745a97bced0f34b8ef64d25dcba8edd34 (2018-11-12 14:20:56 -0500)
wip

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by maiamcc** [comment]
This is the first part of a better solution to #705 -- I'd still love it if the Tiltfile could figure out on load what manifests it defined, but it seems like it'd be even hackier to implement that? @nicks in particular lmk what you think.

File: internal/tiltfile/tiltfile.go [hunk 1]
```diff
@@ -403,7 +407,7 @@ func (t *Tiltfile) globalYaml(thread *skylark.Thread, fn *skylark.Builtin, args
 	return skylark.None, nil
 }
 
-func Load(ctx context.Context, filename string) (*Tiltfile, error) {
+func Load(ctx context.Context, allManifestNames []string, filename string) (*Tiltfile, error) {
```
Comment (nicks) [inline internal/tiltfile/tiltfile.go]: shouldn't this be a []model.ManifestName?

File: internal/engine/buildcontroller.go [hunk 2]
```diff
@@ -147,9 +152,8 @@ func (c *BuildController) logBuildEntry(ctx context.Context, entry buildEntry) {
 	}
 }
 
-func getNewManifestFromTiltfile(ctx context.Context, name model.ManifestName) (model.Manifest, model.YAMLManifest, error) {
-	// Sends any output to the CurrentBuildLog
-	t, err := tiltfile.Load(ctx, tiltfile.FileName)
+func getNewManifestFromTiltfile(ctx context.Context, allManifestNames []string, name model.ManifestName) (model.Manifest, model.YAMLManifest, error) {
```
Comment (nicks) [inline internal/engine/buildcontroller.go]: it seems wrong for the first arg to be strings and the second arg be a ManifestName. shouldn't they both be ManifestNames?

Comment (jazzdan) [inline internal/engine/buildcontroller.go]: why not have `allManifestNames` take a `[]model.ManifestName` and just convert once on the way in when we're parsing the Tiltfile?

**Comment by nicks** [comment]
re: " I'd still love it if the Tiltfile could figure out on load what manifests it defined", are you sure those are the semantics you want? 

for example, if you "tilt up vigoda", i don't think you want to see all the other services.

**Comment by nicks** [comment]
ya, i think that's the right semantics, i don't think we're in disagreement.

**Comment by jazzdan** [comment]
@maiamcc should we close this?

**Comment by maiamcc** [comment]
yep, with new tiltfile, no longer necessary.

**Pull request closed without merging**